platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,204 @@
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private final String errorPagePath = "/ErrorServlet";
+  private ImmutableMap<String, String> errorCodesMap =
+      new ImmutableMap.Builder<String, String>()
+          .put("400", errorPagePath)
+          .put("401", errorPagePath)
+          .put("403", errorPagePath)
+          .put("404", errorPagePath)
+          .put("405", errorPagePath)
+          .put("406", errorPagePath)
+          .put("500", errorPagePath)
+          .put("501", errorPagePath)
+          .put("502", errorPagePath)
+          .put("503", errorPagePath)
+          .put("504", errorPagePath)
+          .build();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  Optional<BundleContext> getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(ErrorPageInjector.class);
+    return (cxfBundle != null) ? Optional.of(cxfBundle.getBundleContext()) : Optional.empty();
+  }
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectErrorPage((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  public void init() {
+    checkForMissedServletContexts();
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkForMissedServletContexts() {
+
+    Optional<BundleContext> optionalBundleContext = getContext();
+    if (!optionalBundleContext.isPresent()) {
+      LOGGER.error(
+          "Problem retrieving the bundle `Platform :: Error :: Page Injector` after it's just been initialized");
+      return; // This should never happen
+    }
+
+    //  Get all registered servletContexts in OSGI
+    final BundleContext context = optionalBundleContext.get();
+    ServiceReference<ServletContext>[] references = null;
+
+    try {
+      references =
+          (ServiceReference<ServletContext>[])
+              context.getAllServiceReferences(
+                  null, "(" + Constants.OBJECTCLASS + "=javax.servlet.ServletContext)");
+
+    } catch (InvalidSyntaxException e) {
+      LOGGER.error(
+          "Problem getting ServletContexts from OSGI, One of the servlets running might print stack traces to the browser. A system restart is recommended. See debug logs for additional details.");
```

Comment:
❗️ Since printing stacktraces to the browser is a security issue, it might not be a good idea to log that in case potential threats view the logs.

And dw, this is coming from a guy who put another vulnerability we had in the title of a ticket 😊.

Reply:
That's a bit of a catch 22... How do you inform the admin without informing potential threats... especially if a log is prepended with Platform-error-page-injector 😅

Reply:
Logging to the log is a-okay. Those files are protected by operating system file controls.

Reply:
@coyotesqrl @stustison